ourjapanlife/findadoc-server
# 28 dockerize server

Resolves #28


# What changed
- [x] Add a Dockerfile to run the server
- [x] Be able to access it from outside the container
- [x] Other build optimization 
- 🏉  Make it talk to the database => Could be broken into another step

# Open questions
- do we always want this to run locally? E.g. can we make two docker-compose workflows, one for a DB only, and another for staging / prod where we want everything in docker?
-  ~GraphQL CodeGen is very chicken / egg... how do we generate types from a running server when we don't have a running server... anyway I don't quite grok this and opted to check in the generated types file for now.~
- What other things should we be doing to ensure a production run?

# Testing instructions

Build the container with:

```sh
docker build . -f ./docker/server/Dockerfile -t findadoc/server
```

Next, 

```sh
docker run -p 49150:3001 -d findadoc/server
```

This will spit out a long container id.

If you have Docker Desktop, you can go see if the container is running. Or you can check the logs like so

```sh
docker logs [container id]
```

### Goal: 

Visit http://localhost:49150 and be able to see the Apollo Explorer
